src/coap_responder.erl

```diff
@@ -233,12 +231,7 @@ handle_unobserve(_ChId, Request, Resource, State) ->
 cancel_observer(#coap_message{options=Options}, State=#state{module=Module, obstate=ObState}) ->
     ok = invoke_callback(Module, coap_unobserve, [ObState]),
     Uri = proplists:get_value(uri_path, Options, []),
-    ok = pg2:leave({coap_observer, Uri}, self()),
-    % will the last observer to leave this group please turn out the lights
```

Comment:
does the new `pg` module does this for us ?

Reply:
there's no `pg:delete/*` function, here's the docs from http://erlang.org/doc/man/pg.html
> Groups are automatically created when any process joins, and are removed when all processes leave the group. Non-existing group is considered empty (containing no processes).

and here's the code to delete the ets table once the last pid left
https://github.com/erlang/otp/blob/master/lib/kernel/src/pg.erl#L468